packages/integrations/src/download-client/nzbget/nzbget-integration.ts

```diff
@@ -0,0 +1,122 @@
+    const history = await nzbGetClient.history();
+    const nzbGetStatus = await nzbGetClient.status();
+    const status: DownloadClientStatus = {
+      paused: nzbGetStatus.DownloadPaused,
+      rates: { down: nzbGetStatus.DownloadRate },
+      type,
+    };
+    const items = queue
+      .map((file): DownloadClientItem => {
+        const state = NzbGetIntegration.getNzbQueueState(file.Status);
+        return {
+          type,
+          id: file.NZBID.toString(),
+          index: file.MaxPriority,
+          name: file.NZBName,
+          size: file.FileSizeLo + file.FileSizeHi * Math.pow(2, 32),
+          downSpeed: file.ActiveDownloads > 0 ? nzbGetStatus.DownloadRate : 0,
+          time: (file.RemainingSizeLo + file.RemainingSizeHi * Math.pow(2, 32)) / (nzbGetStatus.DownloadRate / 1000),
+          added: dayjs().valueOf() - file.DownloadTimeSec * 1000,
+          state,
+          progress: file.DownloadedSizeMB / file.FileSizeMB,
+          category: file.Category,
+        };
+      })
+      .concat(
+        history.map((file, index): DownloadClientItem => {
+          const state = NzbGetIntegration.getNzbHistoryState(file.ScriptStatus);
+          return {
+            type,
+            id: file.NZBID.toString(),
+            index,
+            name: file.Name,
+            size: file.FileSizeLo + file.FileSizeHi * Math.pow(2, 32),
+            time: dayjs().valueOf() - file.HistoryTime * 1000,
+            added: (file.HistoryTime - file.DownloadTimeSec) * 1000,
+            state,
+            progress: 1,
+            category: file.Category,
+          };
+        }),
+      );
+    return { status, items };
+  }
+
+  public async pauseQueueAsync() {
+    await this.getClient().pausedownload();
+  }
+
+  public async pauseItemAsync({ id }: DownloadClientItem): Promise<void> {
+    await this.getClient().editqueue("GroupPause", "", [Number(id)]);
+  }
+
+  public async resumeQueueAsync() {
+    await this.getClient().resumedownload();
+  }
+
+  public async resumeItemAsync({ id }: DownloadClientItem): Promise<void> {
+    await this.getClient().editqueue("GroupResume", "", [Number(id)]);
+  }
+
+  public async deleteItemAsync({ id, progress }: DownloadClientItem, fromDisk: boolean): Promise<void> {
+    const client = this.getClient();
+    if (fromDisk) {
+      const filesIds = (await client.listfiles(Number(id))).map((value) => value.ID);
+      await this.getClient().editqueue("FileDelete", "", filesIds);
+    }
+    if (progress !== 1) {
+      await client.editqueue("GroupFinalDelete", "", [Number(id)]);
+    } else {
+      await client.editqueue("HistoryFinalDelete", "", [Number(id)]);
+    }
+  }
+
+  private getClient() {
+    const url = new URL(this.integration.url);
+    url.username = this.getSecretValue("username");
+    url.password = this.getSecretValue("password");
+    //url.pathname += `${this.getSecretValue("username")}:${this.getSecretValue("password")}`;
```

Comment:
Remove unused code

Reply:
Not unused yet, we need to fix nzbget first to see if we need to change from one format to the other.